### DISCUSSION TITLE
More Indepth explanation for react native

### DISCUSSION THREAD

**Discussion by NexWonka** [WalletConnect/walletconnect-monorepo#1857] [Ideas]
I've notice the react native integration in the docs is very vague, i.e. when you should use sign / auth, how to implement both, more explanation for the code (instead of jumping betwen auth, sign and core).

Can you guys make a react native specific page like the js, kotlin and swift?

**Comment by bkrem** [comment]
cc @crypblizz8 @finessevanes

**Comment by crypblizz8** [comment]
Great thanks for the suggestion @NexWonka. Will get started on this!

**Comment by NexWonka** [comment]
Great, thanks guys! 😁

**Comment by aidankinzett** [comment]
hi, has there been any progress on this? I am trying to use v2 of Wallet Connect for React Native but its very confusing to set up

**Comment by crypblizz8** [comment]
Hi there, our examples for dapp / wallet side is available here: https://github.com/WalletConnect/react-native-examples

Will be publishing more documentation and tutorials on this in the coming week or two.

**Reply by bogdan-marian** [reply]
> Hi there, our examples for dapp / wallet side is available here: https://github.com/WalletConnect/react-native-examples
> 
> Will be publishing more documentation and tutorials on this in the coming week or two.

I just tried to use this example on android and they do not work: 
```
 INFO  {"context": "client"} SignClient Initialization Failure
 ERROR  {"context": "client"} WebSocket connection failed for URL: wss://relay.walletconnect.com?auth=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJkaWQ6a2V5Ono2TWtyYVB1UWh3ZHBMNkVncVFacW50Z1Q4QzV4a0xqZ2NpdnpuVjRZUzlDZWtYMyIsInN1YiI6IjJiNDA4YmJhNmJjOGNhZDMyZmFiOWRmMDhjNzAwOGEyNDE3MmJmMTFmYzY2OTBkYjQzNGIzNDIyNWI1MGJkMTIiLCJhdWQiOiJ3c3M6Ly9yZWxheS53YWxsZXRjb25uZWN0LmNvbSIsImlhdCI6MTY3OTIwOTg5MCwiZXhwIjoxNjc5Mjk2MjkwfQ.F2TtLjijDT3UgKq3URoocbvQ85vRA4rBz5f_cKL89XLkTknuhFn2J5Z2DiIElVtdZJ2noEoCnLYwrt9ACODmBA&ua=wc-2%2Fjs-2.4.0%2Funknown-%2Freact-native
```

**Reply by bkrem** [reply]
@bogdan-marian from the websocket connection string it looks like you're missing the projectId.

cc @crypblizz8 @ignaciosantise can we update the RN examples to the latest Sign/Core versions pls, we've shipped better websocket error messaging for situations like this and should generally update 🙏

**Reply by finessevanes** [reply]
https://github.com/WalletConnect/react-native-examples/pull/55
